joomla/joomla-cms
# [4.0] Build Tools code cleanup

Pull Request for Issue # .

### Summary of Changes
- Asynchronous code everywhere


### Testing Instructions
- Check that all the media files exist as before. A diff tool like http://www.sourcegear.com/diffmerge/ and renaming the media folder after each `npm install` (before applying the PR and after)
- Please post your OS with your result

### Actual result BEFORE applying this Pull Request
Task `npm install` takes x amount of time
<img width="1056" alt="Screenshot 2021-02-05 at 01 41 31" src="https://user-images.githubusercontent.com/3889375/106973722-bfa2dc00-6753-11eb-82d9-7995f4c5b16b.png">



### Expected result AFTER applying this Pull Request
Task `npm install` takes almost half the time
<img width="1030" alt="Screenshot 2021-02-05 at 01 42 09" src="https://user-images.githubusercontent.com/3889375/106973785-da755080-6753-11eb-80e2-dadaeff4aebb.png">



### Documentation Changes Required
No, code clean up

- Needs at least a Windows and a Unix based OS test

@laoneo I think you've asked for this one
